src/tof/result.py

```diff
@@ -0,0 +1,230 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+from itertools import chain
+from typing import Dict, List, Optional, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipp as sc
+from matplotlib.collections import LineCollection
+
+
+from .chopper import Chopper
+from .detector import Detector
+from .pulse import Pulse
+from .utils import Plot
+
+
+class Result:
+    """Result of a simulation."""
+
+    def __init__(
+        self, pulse: Pulse, choppers: Dict[str, Chopper], detectors: Dict[str, Detector]
+    ):
+        self._pulse = pulse
+        self._choppers = choppers
+        self._detectors = detectors
+
+    @property
+    def choppers(self) -> Dict[str, Chopper]:
+        """The choppers in the model."""
+        return self._choppers
```

Comment:
So users can still change the choppers of the result? Isn't that similar to the problem we had before?

Reply:
Yes you are right, I know am returning read-only objects in the result.

Reply:
Can still remove or insert choppers though?

Reply:
Would you then turn `self._choppers` and `self._detectors` into a read-only dict?

Reply:
https://docs.python.org/3/library/types.html#types.MappingProxyType, maybe?